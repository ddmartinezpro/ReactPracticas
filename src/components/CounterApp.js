import React, {useState} from "react";
import PropTypes from 'prop-types'

const CounterApp = ({value = 10}) => {

    const [counter, setCounter] = useState(value);

    const handleAdd = () => {
        setCounter(counter + 1)
    }

    const handleReset = () => {
        setCounter(value)
    }

    const handleSub = () => {
        setCounter(counter - 1)
    }

    return (
        <>
            <h1>Counter App</h1>
            <h2>{counter}</h2>
            <span>Se puede incrementar o decrementar</span>
            <br/>
            <button onClick={handleAdd}>Incrementar</button>
            <button onClick={handleReset}>Reset</button>
            <button onClick={handleSub}>Decrementar</button>
        </>
    );
};

CounterApp.propTypes = {
    value: PropTypes.number,
}

export default CounterApp;
