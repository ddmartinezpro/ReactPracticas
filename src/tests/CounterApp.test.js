import CounterApp from "../components/CounterApp";
import {render, screen, fireEvent} from "@testing-library/react";

describe('Pruebas en CounterApp', () => {
    test('Debe coincidir con el snap y tener por default el contador en 10', () => {
        const { container } = render(<CounterApp />);

        expect(container).toMatchSnapshot();

        // El uso de 'heading' representa las etiquetas 'h', el 'level 2' seria el 'h2'
        const counterNumber = screen.getByRole('heading', { level: 2 });
        expect(counterNumber.textContent).toBe('10')
    });

    test('Si se manda 3 y se aumenta en 1 se debe mostrar 4', () => {
        const { container } = render(<CounterApp value={3}/>);

        // Se simula un click en el boton de incrementar el contador que se referencia mediante su texto
        const targetButton = screen.getByText('Incrementar');
        fireEvent.click(targetButton);

        // El uso de 'heading' representa las etiquetas 'h', el 'level 2' seria el 'h2'
        const counterNumber = screen.getByRole('heading', { level: 2 });
        expect(counterNumber.textContent).toBe('4');

        expect(container).toMatchSnapshot();
    })
});
