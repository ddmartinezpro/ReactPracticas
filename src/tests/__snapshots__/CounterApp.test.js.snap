// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en CounterApp Debe coincidir con el snap y tener por default el contador en 10 1`] = `
<div>
  <h1>
    Counter App
  </h1>
  <h2>
    10
  </h2>
  <span>
    Se puede incrementar o decrementar
  </span>
  <br />
  <button>
    Incrementar
  </button>
  <button>
    Reset
  </button>
  <button>
    Decrementar
  </button>
</div>
`;

exports[`Pruebas en CounterApp Si se manda 3 y se aumenta en 1 se debe mostrar 4 1`] = `
<div>
  <h1>
    Counter App
  </h1>
  <h2>
    4
  </h2>
  <span>
    Se puede incrementar o decrementar
  </span>
  <br />
  <button>
    Incrementar
  </button>
  <button>
    Reset
  </button>
  <button>
    Decrementar
  </button>
</div>
`;
